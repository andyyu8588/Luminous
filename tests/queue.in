// For compiler/VM testing purpose

import Queue

class Object {
    public constructor() {
        print("Object constructed");
    }
}

queue = Queue();
queue.push(Object());
queue.push("Hello world!");
queue.push(3.1415926);
print(queue.isEmpty());
print(queue.size());
print(queue.pop());
print(queue.size());
print(queue.front());
queue.pop();
print(queue.size());
print(queue.pop());
print(queue.isEmpty());
print(queue.size());