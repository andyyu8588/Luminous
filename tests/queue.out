Object constructed
false
3
Object instance
2
Hello world!
1
3.14159
true
0
